### PR TITLE
Fix: Remove `var.region`

### PR DESCRIPTION
**Pull request by korenyoni**
## what

- Remove `var.region` variable.
- Add `count` meta-attributes to data sources.
- Use `aws_region` data source in place of `var.region`.

## why
- Modules do not usually contain `var.region`, only [components](https://github.com/cloudposse/terraform-aws-components) do. 
- Adding a count meta-attribute to all resources and data sources based on `module.this.enabled` is best-practice as it prevents Terraform from populating the Terraform state needlessly if the module is disabled.

## references
- N/A

### CODE REVIEW

**Comment by korenyoni** [comment]
`test/bats` are failing because `terraform validate .` doesn't take into account that the provider is pinned to `3.x`

https://github.com/cloudposse/terraform-aws-mwaa/blob/4c53a71676c1aba98afd453bb9546113580939c4/test/Makefile#L36-L38

```hcl
± |fix/region-var ?:1 ✗| → terraform validate .
╷
│ Error: Unsupported attribute
│ 
│   on .terraform/modules/mwaa_s3_bucket/main.tf line 166, in resource "aws_s3_bucket" "default":
│  166:         for_each = local.s3_replication_rules == null ? [] : local.s3_replication_rules
│ 
│ This object does not have an attribute named "s3_replication_rules".
```

File: main.tf [hunk 1]
```diff
@@ -1,14 +1,22 @@
-data "aws_caller_identity" "current" {}
-data "aws_partition" "current" {}
+data "aws_caller_identity" "current" {
+  count = local.enabled ? 1 : 0
+}
+data "aws_partition" "current" {
```
Comment (aknysh) [inline main.tf:4]: add empty lines b/w resources

File: main.tf [hunk 2]
```diff
@@ -1,14 +1,22 @@
...
   enabled = module.this.enabled
 
   security_group_enabled = local.enabled && var.create_security_group
   s3_bucket_enabled      = local.enabled && var.create_s3_bucket
   iam_role_enabled       = local.enabled && var.create_iam_role
-  account_id             = data.aws_caller_identity.current.account_id
-  partition              = data.aws_partition.current.partition
+  account_id             = join("", data.aws_caller_identity.current.*.account_id)
+  partition              = join("", data.aws_partition.current.*.partition)
+  region                 = join("", data.aws_region.current.*.name)
```
Comment (aknysh) [inline main.tf:19]: why not use `local.region` is we have it?

**Review by Nuru — CHANGES_REQUESTED** [review]
Leave `var.region` in place. It is a standard required variable for all AWS components and is implied for all AWS modules. The only reason we do not see it more often in modules is that it is usually enough that it is implicitly provided by the AWS provider.

**Pull request closed without merging**